docker: openwrt: bump prplwrt version

arnout opened the pull request:
The prplwrt branch at git.prpl.dev:prplmesh/prplwrt was update to
re-sync with upstream prplwrt. Make sure we use the new version.

This will fix build errors that people see when doing a clean build,
without any docker cache. The problem is (probably) that the package
feeds don't have hashes, so we're using the latest for them rather than
the ones corresponding to our prplwrt fork. In CI this issue doesn't
occur because the feeds are taken from the docker cache.

arnout committed:
docker: openwrt: bump prplwrt version

The prplwrt branch at git.prpl.dev:prplmesh/prplwrt was update to
re-sync with upstream prplwrt. Make sure we use the new version.

This will fix build errors that people see when doing a clean build,
without any docker cache. The problem is (probably) that the package
feeds don't have hashes, so we're using the latest for them rather than
the ones corresponding to our prplwrt fork. In CI this issue doesn't
occur because the feeds are taken from the docker cache.

Signed-off-by: Arnout Vandecappelle (Essensium/Mind) <arnout@mind.be>
Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
feed_prpl: enable sta_ifaces in default config

Bump up feed_prpl to the following commit (unmerged):
c9f9407 - Arnout Vandecappelle (Essensium/Mind), 3 days ago : prplmesh: enable sta_ifaces in default config

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

VladyslavTupikin commented:
Turris Omnia build error:

```
Cloning into './feeds/prpl'...
 fatal: 'dev/prplmesh-enable-wireless-backhaul' is not a commit and a branch 'dev/prplmesh-enable-wireless-backhaul' cannot be created from it
 failed.
```

Glinet build error:

```
 Cloning into './feeds/prpl'...
 fatal: 'dev/prplmesh-enable-wireless-backhaul' is not a commit and a branch 'dev/prplmesh-enable-wireless-backhaul' cannot be created from it
 failed.
```
Teest on RAX40 failure:

```
 pexpect.exceptions.TIMEOUT: Timeout exceeded.
 <pexpect.fdpexpect.fdspawn object at 0x7f0153dc3e48>
 searcher: searcher_re:
     0: re.compile(b'GRX500 #')
```

Commit:
docker:builder: openwrt wireless bh support

1. Bump openwrt hush to add wireless bh support in prplwrt.
2. Stop using profile_feeds/netgear0rax40.yml since now prplwrt already
includes prplmesh support with fixed hashes.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
docker:builder:openrt: cleanups

- Since we don't use yq anymore, remove it from docker
- remove unused profile_feeds

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
docker: builder: prplmesh build.sh clean all

Cleanup prplmesh folder is not complete since removing the build dir is
not enough when it was built with USE_SOURCE_DIR natively.
Add cleanup to delete all openwrt artifacts.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

tomereli commented:
> Turris Omnia build error:
> 
> ```
> Cloning into './feeds/prpl'...
>  fatal: 'dev/prplmesh-enable-wireless-backhaul' is not a commit and a branch 'dev/prplmesh-enable-wireless-backhaul' cannot be created from it
>  failed.
> ```
> 
> Glinet build error:
> 
> ```
>  Cloning into './feeds/prpl'...
>  fatal: 'dev/prplmesh-enable-wireless-backhaul' is not a commit and a branch 'dev/prplmesh-enable-wireless-backhaul' cannot be created from it
>  failed.
> ```
> 
> Teest on RAX40 failure:
> 
> ```
>  pexpect.exceptions.TIMEOUT: Timeout exceeded.
>  <pexpect.fdpexpect.fdspawn object at 0x7f0153dc3e48>
>  searcher: searcher_re:
>      0: re.compile(b'GRX500 #')
> ```

Fixed

arnout commented on tools/docker/builder/openwrt/build.sh:
```diff
@@ -172,8 +172,9 @@ main() {
-OPENWRT_VERSION='bd19f9ab26ad234b6f10cce23cd0dc41b9371929'
-PRPL_FEED='https://git.prpl.dev/prplmesh/feed-prpl.git^53d1e11003ce318c043c42063bbd2f57d15aac81'
+OPENWRT_VERSION='add783ec25be542fdf3ba414a6b08345b5d76a5e'
+# TODO use hash instead of branch
```
This TODO is no longer correct because there is an actual hash used here...

arnout commented on tools/docker/builder/openwrt/scripts/build.sh:
```diff
@@ -4,8 +4,10 @@
 cp -r /home/openwrt/prplMesh_source /home/openwrt/prplMesh
 # We want to make sure that we do not keep anything built from the host:
-rm -rf /home/openwrt/prplMesh/build
+cd /home/openwrt/prplMesh && \
+    rm -rf build ipkg-* .built* .configured* .pkgdir .prepared .quilt_checked .source_dir
 
+cd /home/openwrt/openwrt
```
(nitpick)

I prefer
```suggestion
(cd /home/openwrt/prplMesh && \
    rm -rf build ipkg-* .built* .configured* .pkgdir .prepared .quilt_checked .source_dir)
```

arnout reviewed (commented):
The first two commits are no longer needed, they're basically superseded by the third one. Well, not for turris and glinet, but we're a bit inconsistent there...

I don't really like removing the profile, because it means that whenever we need to change something in one of the feeds, e.g. dwpal, we have to make 7 pull requests:
1. in our dwpal fork;
2. in our dwpal-feed fork, to update dwpal - this has to be updated when 1 is merged;
3. in our prplwrt fork to update dwpal-feed in profile - this has to be updated when 2 is merged, and also when 1 is merged;
4. in prplmesh to update prplwrt - this has to be updated when 3 is merged, and also when 2 is merged, and also when 1 is merged;
5. in upstream dwpal to get it upstream;
6. in upstream dwpal-feed to get it upstream - but this one we can drop assuming that it will happen automatically by prplwrt maintainers.
7. in upstream prplwrt to get it upstream - but this one we can drop assuming that it will happen automatically by prplwrt maintainers.

We could merge 1 and 5 by working directly in a pull request in upstream dwpal (i.e. make a pull request that doesn't merge into master of our fork, but directly into upstream master), but that's assuming that upstream dwpal is reactive. Same for 2 and 6, and maybe also for 3 and 7.

The profiles directory allows us to get rid of step 3.

Commit:
build:docker:openwrt remove workarounds for building prplwrt

Now these are fixed in prplmesh prplwrt fork itself so are no longer
needed. Bump the prplwrt hash to include all fixes.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

Commit:
tools:docker:builder: openwrt bump prplwrt hash for ebtables package

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>
